### PR TITLE
Fix S2K params in key unlock

### PR DESCRIPTION
**Pull request by wussler**
Alternative version of https://github.com/ProtonMail/crypto/pull/63

### CODE REVIEW

File: openpgp/packet/private_key.go [hunk 1]
```diff
@@ -404,7 +404,18 @@ func (pk *PrivateKey) Decrypt(passphrase []byte) error {
 		data = data[:len(data)-2]
 	}
 
-	return pk.parsePrivateKey(data)
+	err := pk.parsePrivateKey(data)
+	if err != nil {
+		return err
+	}
+
+	if !pk.Dummy() {
```
Comment (twiss) [inline openpgp/packet/private_key.go:412]: I think `pk.Dummy()` isn't possible at this point (see the beginning of the function)

File: openpgp/packet/private_key.go [hunk 2]
```diff
@@ -404,7 +404,18 @@ func (pk *PrivateKey) Decrypt(passphrase []byte) error {
...
+		pk.s2kType = S2KNON
+		pk.s2k = nil
+		pk.Encrypted = false
```
Comment (twiss) [inline openpgp/packet/private_key.go:415]: All of the functions that `parsePrivateKey` calls already set `pk.Encrypted = false`, so this is a bit redundant; that being said, I agree that this is the more logical place to do it, so you could also remove all of those. But as you prefer :)

File: openpgp/packet/private_key.go [hunk 3]
```diff
@@ -221,21 +222,23 @@ func (pk *PrivateKey) parse(r io.Reader) (err error) {
...
 	} else {
-		pk.encryptedData, err = ioutil.ReadAll(r)
+		privateKeyData, err = ioutil.ReadAll(r)
 		if err != nil {
 			return
 		}
 	}
 	if !pk.Encrypted {
-		return pk.parsePrivateKey(pk.encryptedData)
+		pk.encryptedData = nil
```
Comment (twiss) [inline openpgp/packet/private_key.go:237]: I'm pretty sure it's already nil at this point, I don't think you need to set this here

Comment (wussler) [reply openpgp/packet/private_key.go]: I checked, fair enough, applied.
I don't love relying on implicit defaults but OK.

**Review by twiss — APPROVED** [review]

**Pull request merged**